### DISCUSSION TITLE
Loader function is being executed on the client side

### DISCUSSION THREAD

**Discussion by parsagholipour** [remix-run/remix#6848] [Q&A] [answered]
I have the following code in root.tsx
```js
import {PrismaClient} from '@prisma/client';

export async function loader({request, context}: LoaderArgs) {
  const prisma = new PrismaClient();
...
```

and I get the following error
`Error: PrismaClient is unable to be run in the browser.`
Basically I cannot use any library that cannot be executed in the browser.

**Comment by kiliman** [comment]
Prisma should be exported from a *.server file. This is due to a side effect of importing prisma in the route file. 

For an example see https://remix.run/docs/en/main/tutorials/jokes#connect-to-the-database

**Reply by parsagholipour** [reply]
I did this but I still get the same error
```js
import {PrismaClient} from './lib/db.server';
export async function loader({request, context}: LoaderArgs) {
  const prisma = new PrismaClient();
...
```

`db.server.tsx` :

```js
import {PrismaClient as _PrismaClient} from '@prisma/client';

export const PrismaClient = _PrismaClient;
```

**Reply by lili21** [reply]
how about try it this way

```
// loader.server.ts
export const loader = xxxx

// your page
export { loader } from './loader.server'
```

**Reply by parsagholipour** [reply]
@lili21 Same

**Reply by lili21** [reply]
can you provide a reproduce repo?

**Comment by parsagholipour** [comment, marked as answer]
The reason was it was being run on Oxygen server instead of Express js. node APIs are not available in the environment.
